docs/WritingTests.asciidoc

```diff
@@ -22,52 +22,208 @@ https://github.com/os-autoinst/openQA[official repository].
 == Basic
 [id="basic"]
 
-This section explains the basic layout of openQA tests and the API available in tests.
-openQA tests are written in the *Perl* programming language. Some basic but no
-in-depth knowledge of Perl is needed. This document assumes that the reader
-is already familiar with Perl.
+This section explains the basic layout of openQA test modules and the API
+available. Test modules are written in the *Perl* programming language.
+However there is support for the *Python* programming language (through the Perl module
+`Inline::Python`).
 
-== API
+This document assumes that the reader is already familiar with Perl or Python.
+
+== Test module API
 [id="api"]
 :testapi: https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm[os-autoinst]
 
-{testapi} provides the API for the tests using the os-autoinst backend, you can
-take a look to the published documentation at http://open.qa/api/testapi/. This test API
-is sometimes also referred to as an openQA DSL, because in some contexts it can look like
-a domain specific language.
+{testapi} provides the API for the test modules using the os-autoinst backend.
+Take a look at the http://open.qa/api/testapi[test API documentation] for further
+information. Note that this test API is sometimes also referred to as an
+openQA DSL, because in some contexts it can look like a domain specific
+language.
+
+=== Notes on the Python API
+[id="notes-python-api"]
+
+The Python integration that OpenQA offers through `Inline::Python` also allows
+the test modules to import other Perl modules with the usage of the `perl`
```

Comment:
This whole block should be much further below. First a basic example should be featured as we had in before, only much later you should add this note

Reply:
Agreed, I didn't quite feel a good place for it, but wanted the notes to be somewhere close. I'll move it down the first basic explanation, maybe there makes more sense.